core/src/main/scala/tech/beshu/ror/accesscontrol/audit/DataStreamAuditSinkCreator.scala

```diff
@@ -0,0 +1,175 @@
+import tech.beshu.ror.accesscontrol.audit.DataStreamAuditSinkCreator.DataStreamSettings
+import tech.beshu.ror.accesscontrol.domain.{DataStreamName, TemplateName}
+import tech.beshu.ror.audit.instances.FieldType
+import tech.beshu.ror.es.DataStreamService
+import tech.beshu.ror.implicits.*
+import ujson.Value
+
+import scala.concurrent.duration.*
+
+final class DataStreamAuditSinkCreator(services: NonEmptyList[DataStreamService]) extends Logging {
+
+  def createIfNotExists(streamSettings: DataStreamSettings): Task[Unit] = {
+    services.toList.traverse(createIfNotExists(_, streamSettings)).map((_: List[Unit]) => ())
+  }
+
+  private def createIfNotExists(service: DataStreamService, settings: DataStreamSettings): Task[Unit] = {
+    service
+      .checkDataStreamExists(settings.dataStreamName)
+      .flatMap {
+        case true =>
+          Task.delay(logger.info(s"Data stream ${settings.dataStreamName.show} already exists"))
+        case false =>
+          setupDataStream(service, settings)
+      }
+  }
+
+  private def setupDataStream(service: DataStreamService, settings: DataStreamSettings): Task[Unit] = {
+    for {
+      _ <- Task.delay(logger.info(s"Trying to setup ROR audit sink with data stream ${settings.dataStreamName.show}.."))
+      _ <- service.createIndexLifecyclePolicy(settings.lifecyclePolicy, settings.lifecyclePolicyJson(service.capabilities.ilmMaxPrimaryShardSize))
+      _ <- service.createComponentTemplateForMappings(settings.componentTemplateMappingsId, settings.mappings, settings.mappingsMetadata)
+      _ <- service.createComponentTemplateForIndex(settings.componentTemplateSettingsId, settings.lifecyclePolicy, settings.indexSettingsMetadata)
+      _ <- service.createIndexTemplate(settings.indexTemplate, settings.dataStreamName, NonEmptyList.of(settings.componentTemplateMappingsId, settings.componentTemplateSettingsId), settings.indexTemplateMetadata)
+      _ <- service.createDataStream(settings.dataStreamName)
+      _ <- Task.delay(logger.info(s"ROR audit data stream ${settings.dataStreamName.show} created."))
+    } yield ()
+  }
+}
+
+
+object DataStreamAuditSinkCreator {
+
+  final case class DataStreamSettings(dataStreamName: DataStreamName.Full,
+                                      documentMappings: Map[String, FieldType]) {
+    def lifecyclePolicy: String = s"${dataStreamName.value.value}-lifecycle-policy"
+
+    def componentTemplateSettingsId: TemplateName = templateNameFrom(s"${dataStreamName.value.value}-settings")
+
+    def componentTemplateMappingsId: TemplateName = templateNameFrom(s"${dataStreamName.value.value}-mappings")
+
+    def indexTemplate: TemplateName = templateNameFrom(s"${dataStreamName.value.value}-template")
+
+    private def templateNameFrom(value: String) = {
+      TemplateName
+        .fromString(value)
+        .getOrElse(throw new IllegalStateException("Template name should be non-empty"))
+    }
+
+    val mappings: Value = serializeMappings(documentMappings)
```

Comment:
AFAIR in case of the ROR audit index we use dynamic mapping, don't we? 
Shouldn't we do it in the data stream case in the same way?